redis/cache.py

```diff
@@ -0,0 +1,401 @@
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+from enum import Enum
+from typing import Any, List, Optional, Union
+
+
+class CacheEntryStatus(Enum):
+    VALID = "VALID"
+    IN_PROGRESS = "IN_PROGRESS"
+
+
+class EvictionPolicyType(Enum):
+    time_based = "time_based"
+    frequency_based = "frequency_based"
+
+
+class CacheKey:
+    def __init__(self, command: str, redis_keys: tuple):
+        self.command = command
+        self.redis_keys = redis_keys
+
+    def get_redis_keys(self) -> tuple:
+        return self.redis_keys
+
+    def __hash__(self):
+        return hash((self.command, self.redis_keys))
+
+    def __eq__(self, other):
+        return hash(self) == hash(other)
```

Comment:
You can simplify the code with frozen dataclass:
```
from dataclasses import dataclass

@dataclass(frozen=True)
class CacheKey:
    command: str
    redis_keys: tuple
```
The same applies to CacheEntry below.

Reply:
Fixed for CacheKey, CacheEntry isn't read-only cannot apply the same